src/components/uv-scroll.js

```diff
@@ -1,20 +1,79 @@
+const textureToData = new Map();
+const registeredTextures = [];
+
+export class UVScrollSystem {
+  tick(_t, dt) {
+    for (let i = 0; i < registeredTextures.length; i++) {
+      const map = registeredTextures[i];
+      const { offset, instances } = textureToData.get(map);
+      const { component } = instances[0];
+
+      offset.addScaledVector(component.data.speed, dt / 1000);
+
+      const increment = component.data.increment;
+      map.offset.x = increment.x ? offset.x - (offset.x % increment.x) : offset.x;
```

Comment:
Today I learned that `%` works on all numbers, not just integers. (I guess this makes sense, given that javascript's "integers" are just those `Number`s that are not `NaN`, `+∞`, or `-∞`, and satisfies `floor(abs(argument)) == abs(argument)` https://tc39.es/ecma262/#sec-isinteger )